### PR TITLE
chore(): package.json properly expose TS typings for dual packages

### PR DESCRIPTION
**Pull request by thekip**
fixes: https://github.com/lingui/js-lingui/issues/1363

### CODE REVIEW

**Comment by andrii-bodnar** [comment]
I would suggest renaming the `e2e-tests` -> `tests` and creating inside a separate directory `e2e`. In case some new test types are added in the future

File: e2e-tests/typescript-nodenext-resolution/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (andrii-bodnar) [inline e2e-tests/typescript-nodenext-resolution/package.json:1]: what about `"private": true`?

Comment (thekip) [reply e2e-tests/typescript-nodenext-resolution/package.json:1]: it doesn't matter, lerna setup to publish only from `./packages`

Comment (andrii-bodnar) [reply e2e-tests/typescript-nodenext-resolution/package.json:1]: anyway, make it consistent. The `e2e-tests/node-api/package.json` has this flag

**Comment by thekip** [comment]
> I would suggest renaming the e2e-tests -> tests and creating inside a separate directory e2e. In case some new test types are added in the future

I would rather rename it to "sandboxes", or something like this. Because they are a not a test in a basic meaning, they are also involved in yarn workspaces install mechanism etc. 
So creating a subfolder may make this all setup is too complicated and confusing.

File: e2e-tests/typescript-nodenext-resolution/readme.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Check support typescript's NodeNext moduleResolution with dual `@lingui` packages
+
+https://github.com/microsoft/TypeScript/issues/50466
```
Comment (andrii-bodnar) [inline e2e-tests/typescript-nodenext-resolution/readme.md:3]: move it to the `index.ts` as a comment, this readme doesn't make a lot of sense

**Comment by andrii-bodnar** [comment]
sandboxes will confuse even more. I mean something like here - https://github.com/ionic-team/stencil/tree/main/test

**Comment by thekip** [comment]
Lets rename it to test, but without additional level. Each folder in that directory - different test case, which might be e2e or any other type

**Comment by thekip** [comment]
Done. I also changed version constaint to "*" because lerna will not bump versions in these folders, but we should rely on the latest available in monorepo.

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**